prime-dotnet-webapi/Services/EnrolleeProfileVersionService.cs

```diff
@@ -0,0 +1,59 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Http;
+using Microsoft.EntityFrameworkCore;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using Newtonsoft.Json.Serialization;
+using Prime.Models;
+
+namespace Prime.Services
+{
+    public class EnrolleeProfileVersionService : BaseService, IEnrolleeProfileVersionService
+    {
+        private JsonSerializer _camelCaseSerializer = JsonSerializer.Create(
+            new JsonSerializerSettings
+            {
+                ContractResolver = new CamelCasePropertyNamesContractResolver()
+            }
+        );
+
+        public EnrolleeProfileVersionService(
+            ApiDbContext context,
+            IHttpContextAccessor httpContext
+            ) : base(context, httpContext)
+        { }
+
+        public async Task<IEnumerable<EnrolleeProfileVersion>> GetEnrolleeProfileVersionsAsync(int enrolleeId)
+        {
+            return await _context.EnrolleeProfileVersions
+                .Where(epv => epv.EnrolleeId == enrolleeId)
+                .ToListAsync();
+        }
+
+        public async Task<EnrolleeProfileVersion> GetEnrolleeProfileVersionAsync(int enrolleeProfileVersionId)
+        {
+            return await _context.EnrolleeProfileVersions
+                .Where(epv => epv.Id == enrolleeProfileVersionId)
+                .FirstOrDefaultAsync();
```

Comment:
.First(OrDefault) and .Single(OrDefault) have overloads that take a selector for this exact case when you are doing a simple where followed by the fetching statement. This should be `.FirstOrDefaultAsync(epv => epv.Id == enrolleeProfileVersionId)`

Reply:
I also assume this should be SingleOrDefault, since ids are unique